runtime/registry.go

```diff
@@ -22,36 +22,14 @@ func (r *Runtime) FindInstance(ctx context.Context, instanceID string) (*drivers
 }
 
 func (r *Runtime) CreateInstance(ctx context.Context, inst *drivers.Instance) error {
-	// Check OLAP connection
-	olap, _, err := r.checkOlapConnection(inst)
+	_, err := r.connectorDef(inst, inst.OLAPDriver)
 	if err != nil {
-		return err
-	}
-	defer olap.Close()
-
-	// Check repo connection
-	repo, _, err := r.checkRepoConnection(inst)
-	if err != nil {
-		return err
+		return fmt.Errorf("invalid olap driver")
 	}
-	defer repo.Close()
 
-	// Check that it's a driver that supports embedded catalogs
-	if inst.EmbedCatalog {
-		_, ok := olap.AsCatalogStore(inst.ID)
-		if !ok {
-			return errors.New("driver does not support embedded catalogs")
-		}
-	}
-
-	// Prepare connections for use
-	err = olap.Migrate(ctx)
-	if err != nil {
-		return fmt.Errorf("failed to prepare instance: %w", err)
-	}
-	err = repo.Migrate(ctx)
+	_, err = r.connectorDef(inst, inst.RepoDriver)
 	if err != nil {
-		return fmt.Errorf("failed to prepare instance: %w", err)
+		return fmt.Errorf("invalid repo driver")
 	}
 
 	// this is a hack to set variables and pass to connectors
```

Comment:
Can this be removed now and populated at runtime based on `r.opts.AllowHostAccess`?

Reply:
The older reconcile still doesn't use `AcquireHandle` and opens connections directly. Once we move to new reconcile it can be removed.